src/state.rs

```diff
@@ -0,0 +1,58 @@
+//! Program state accounts
+
+use {
+    solana_program::clock::UnixTimestamp, solana_program::pubkey::Pubkey,
+    spl_pod::optional_keys::OptionalNonZeroPubkey,
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum AccountType {
+    Uninitialized,
+    Config,
+    Stake,
+}
```

Comment:
Just curious; I don't see this used anywhere yet. Something akin to Token-2022 account state?

Reply:
It's essentially just a small account discriminator, but we should probably use `spl-discriminate` for these

Reply:
Up to you on the `spl-discriminate` bit. Saving 7 bytes isn't bad though!

Reply:
Might be easier to use `spl-discriminate` since the types are 8-byte aligned already.

Reply:
can we leave it for follow-up work?